client/landing/stepper/declarative-flow/internals/steps-repository/import/helper.ts

```diff
@@ -65,16 +67,34 @@ export function generateStepPath( stepName: string, stepSectionName?: string ) {
 }
 
 export async function addTempSiteToSourceOption( targetBlogId: number, sourceSiteSlug: string ) {
-	return wpcom.req
-		.post( {
-			path: `/migrations/from-source/${ sourceSiteSlug }`,
+	try {
+		const sourceSite = await getSite( sourceSiteSlug );
+		const result = await wpcom.req.post( {
+			path: `/migrations/from-source/${ sourceSite.ID }`,
 			apiNamespace: 'wpcom/v2',
 			body: {
 				target_blog_id: targetBlogId,
 			},
-		} )
-		.catch( ( error: Error ) => {
-			// eslint-disable-next-line no-console
-			console.error( 'Unable to store option in source site', error );
 		} );
+		return result;
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error( 'Unable to store option in source site', error );
+	}
+}
+
+export function getSite( sourceSiteSlug: string ) {
+	return wpcomRequest< SiteDetails >( {
+		path: '/sites/' + encodeURIComponent( sourceSiteSlug as string ),
+		apiVersion: '1.1',
+	} );
+}
+
+export function getSourceSiteMigrationData(
+	sourceId: number
+): Promise< SourceSiteMigrationDetails > {
+	return wpcom.req.get( {
+		path: '/migrations/from-source/' + encodeURIComponent( sourceId as number ),
+		apiNamespace: 'wpcom/v2',
+	} );
```

Comment:
I wonder if this is a proper place to add fetching data logic?

Probably, the more appropriate place is somewhere in `/client/data/site`.
There, we can consume `useQuery` hook and follow the standardized approach in the `/client/data` layer.

For example:
```
function useSiteQuery( siteSlug: string ) {
	return useQuery< Site >(
		[ 'site-object', siteSlug ],
		() => wpcom.req.get( { path: `/sites/${ siteSlug }`, apiVersion: '1.1' } )
	);
}
```